source/Nevermore/RelationalTransaction.cs

```diff
@@ -678,6 +678,25 @@ public override string ToString()
             return $"{CreatedTime} - {connection?.State} - {name}";
         }
 
+        string QueryInfoForLogging(string query, CommandParameterValues parameterValues)
+        {
+            return QueryInfoForLogging(query, parameterValues?.Select(p => (p.Key, p.Value)).ToList() ?? new List<(string Key, object Value)>());
+        }
+        
+        string QueryInfoForLogging(IDbCommand command)
+        {
+            return QueryInfoForLogging(command.CommandText, command.Parameters.Cast<IDataParameter>().Select(p => (p.ParameterName, p.Value)).ToList());
+        }
+
+        string QueryInfoForLogging(string query, IReadOnlyList<(string ParameterName, object ParameterValue)> parameters)
+        {
+            if (!parameters.Any())
+                return query;
+
+            var parameterInfo = "Parameters:\r\n" + string.Join(",\r\n", parameters.Select(p => p.ParameterName + ": " + p.ParameterValue));
```

Comment:
Not sure if we should truncate this list so we only print the first X parameters (can have up to 2100)